### ISSUE TITLE
Investigate recipe for spring-sleuth -> micrometer transition

### ISSUE THREAD

**Issue by sambsnyd** [openrewrite/rewrite-spring#299]
Recent versions of spring have moved from spring-sleuth to micrometer for instrumentation. I don't know the details, but there may be opportunity for automation in this area

**Comment by knutwannheden** [comment]
Yes, closer investigation reveals that there are quite a few differences at the lower level. However, I wonder how much of this should be covered by this migration recipe. I suspect that applications using the API are rather rare and those which do probably only use a small subset of the entire API. I propose to start with a recipe covering the main APIs and then try to validate the recipe on the public SaaS tenant against some projects using these APIs. What do you think, @sambsnyd ?

**Comment by knutwannheden** [comment]
Depending on the exact version of Spring Cloud Sleuth the APIs are unfortunately also rather different. With the latest version they tried to align the APIs so that the migration to Micrometer Tracing would be as painless as possible.

Instead of implementing all possible version upgrades or anticipating some specific migration path, the current state of the recipes simply migrates from the latest [Spring Cloud Sleuth v3.1.7](https://www.javadoc.io/doc/org.springframework.cloud/spring-cloud-sleuth-api/latest/index.html) to [Micrometer Tracing v1.0.3](https://javadoc.io/doc/io.micrometer/micrometer-tracing/latest/index.html).
